gst/nnstreamer/tensor_query/nnstreamer_edge.h

```diff
@@ -0,0 +1,165 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2022 Gichan Jang <gichan2.jang@samsung.com>
+ *
+ * @file   nnstreamer_edge.h
+ * @date   25 Mar 2022
+ * @brief  Common library to support communication among devices.
+ * @see    https://github.com/nnstreamer/nnstreamer
+ * @author Gichan Jang <gichan2.jang@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ * @note This file will be moved to nnstreamer-edge repo. (https://github.com/nnstreamer/nnstreamer-edge)
+ */
+
+#ifndef __NNSTREAMER_EDGE_H__
+#define __NNSTREAMER_EDGE_H__
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @brief The maximum number of data instances that nnstreamer-edge data may have.
+ */
+#define NNS_EDGE_DATA_LIMIT (16)
+
+typedef void *nns_edge_h;
+typedef void *nns_edge_event_h;
+typedef void *nns_edge_data_h;
+
+/**
+ * @brief Enumeration for the error codes of nnstreamer-edge.
+ * @todo define error code later
+ */
+typedef enum
+{
+  NNS_EDGE_ERROR_NONE = 0,
+  NNS_EDGE_ERROR_UNKNOWN = -1,
+  NNS_EDGE_ERROR_INVALID_PARAMETER = -2,
+  NNS_EDGE_ERROR_OUT_OF_MEMORY = -3,
+  NNS_EDGE_ERROR_CONNECTION_FAILURE = -4,
+  NNS_EDGE_ERROR_IO = -5,
+} nns_edge_error_e;
```

Comment:
You may use Linux standard error code instead. (errno.h)

Reply:
Having non Linux standard error code will cause some problems in the future. Please reuse Linux error code unless you have some specific reason not to.

Reply:
I added some comment for this issue.
* https://github.com/nnstreamer/nnstreamer/pull/3710#discussion_r865488072